xbmc/video/dialogs/GUIDialogVideoVersion.cpp

```diff
@@ -0,0 +1,1008 @@
+    title = StringUtils::Format("{} ({})", title, year);
+
+  SET_CONTROL_LABEL(CONTROL_LABEL_TITLE,
+                    StringUtils::Format(g_localizeStrings.Get(!m_playMode ? 40022 : 40023), title));
+
+  // bind primary and extras version lists
+  CGUIMessage msg1(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LIST_PRIMARY_VERSION, 0, 0,
+                   m_primaryVideoVersionList.get());
+  OnMessage(msg1);
+
+  CGUIMessage msg2(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LIST_EXTRAS_VERSION, 0, 0,
+                   m_extrasVideoVersionList.get());
+  OnMessage(msg2);
+
+  // disable buttons that need a selected video version
+  CONTROL_DISABLE(CONTROL_BUTTON_REMOVE);
+  CONTROL_DISABLE(CONTROL_BUTTON_SET_DEFAULT);
+
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_BUTTON_ADD_VERSION, !m_playMode);
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_BUTTON_ADD_EXTRAS, !m_playMode);
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_BUTTON_CHOOSE_ART, !m_playMode);
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_BUTTON_RENAME, !m_playMode);
+  CONTROL_ENABLE_ON_CONDITION(CONTROL_BUTTON_PLAY, !m_playMode);
+
+  CGUIDialog::OnInitWindow();
+}
+
+void CGUIDialogVideoVersion::ClearVideoVersionList()
+{
+  CGUIMessage msg1(GUI_MSG_LABEL_RESET, GetID(), CONTROL_LIST_PRIMARY_VERSION);
+  OnMessage(msg1);
+  m_primaryVideoVersionList->Clear();
+
+  CGUIMessage msg2(GUI_MSG_LABEL_RESET, GetID(), CONTROL_LIST_EXTRAS_VERSION);
+  OnMessage(msg2);
+  m_extrasVideoVersionList->Clear();
+}
+
+void CGUIDialogVideoVersion::RefreshVideoVersionList()
+{
+  // clear current version list
+  ClearVideoVersionList();
+
+  const int dbId = m_videoItem->GetVideoInfoTag()->m_iDbId;
+  MediaType mediaType = m_videoItem->GetVideoInfoTag()->m_type;
+  VideoDbContentType itemType = m_videoItem->GetVideoContentType();
+
+  // get primary version list
+  m_database.GetVideoVersions(itemType, dbId, *m_primaryVideoVersionList,
+                              VideoVersionItemType::PRIMARY);
+  m_primaryVideoVersionList->SetContent(CMediaTypes::ToPlural(mediaType));
+
+  // refresh primary version list
+  CGUIMessage msg1(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LIST_PRIMARY_VERSION, 0, 0,
+                   m_primaryVideoVersionList.get());
+  OnMessage(msg1);
+
+  // get extras version list
+  m_database.GetVideoVersions(itemType, dbId, *m_extrasVideoVersionList,
+                              VideoVersionItemType::EXTRAS);
+  m_extrasVideoVersionList->SetContent(CMediaTypes::ToPlural(mediaType));
+
+  // refresh extras version list
+  CGUIMessage msg2(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LIST_EXTRAS_VERSION, 0, 0,
+                   m_extrasVideoVersionList.get());
+  OnMessage(msg2);
+
+  // update default video version
+  m_database.GetDefaultVideoVersion(itemType, dbId, *m_selectedVideoVersion.get());
+}
+
+void CGUIDialogVideoVersion::SetVideoItem(const std::shared_ptr<CFileItem>& item, bool playMode)
+{
+  if (item == nullptr || !item->HasVideoInfoTag() ||
+      item->GetVideoInfoTag()->m_type != MediaTypeMovie)
+    return;
```

Comment:
Log an error/warning? I ask for this in previous reviews. Maybe you overlooked this?

Reply:
Updated. Not sure why it's not labeled correctly.